File: Algorithms/src/1523. Count Odd Numbers in an Interval Range/Code/solution.py
class Solution:
    def countOdds(self, low: int, high: int) -> int:
        # Helper: count odd numbers from 1 to x
        def odds_up_to(x: int) -> int:
            # // is integer division (floor)
            return (x + 1) // 2
        
        # Odds in [low, high]
        return odds_up_to(high) - odds_up_to(low - 1)

File: Algorithms/src/1523. Count Odd Numbers in an Interval Range/Code/solution.cpp
class Solution
{
public:
    int countOdds(int low, int high)
    {
        // Helper lambda: count of odd numbers from 1 to x
        auto oddsUpTo = [](int x) -> int
        {
            // Integer division automatically takes floor
            return (x + 1) / 2;
        };

        // Odds in [low, high] = oddsUpTo(high) - oddsUpTo(low - 1)
        return oddsUpTo(high) - oddsUpTo(low - 1);
    }
};
